plugins/org.jkiss.dbeaver.erd.ui/src/org/jkiss/dbeaver/erd/ui/part/AssociationPart.java

```diff
@@ -192,7 +195,12 @@ protected void setConnectionStyles(PolylineConnection conn) {
 
         conn.setLineWidth(2);
         if (!identifying || constraintType.isLogical()) {
-            conn.setLineStyle(SWT.LINE_CUSTOM);
+            final DBPPreferenceStore store = ERDUIActivator.getDefault().getPreferences();
+            if (store.getString(ERDUIConstants.PREF_ROUTING_TYPE).equals(ERDUIConstants.ROUTING_MIKAMI)) {
+                conn.setLineStyle(SWT.LINE_DOT);
+            } else {
+                conn.setLineStyle(SWT.LINE_CUSTOM);
+            }
```

Comment:
I don't like that it's hardcoded just for specific router.

Can we abstract these things into a dedicated "Decorator" class?